### PR TITLE
Add rvc example app

### LINKED ISSUE #28564: Add RVC Device Example App
There should be a fully functional example of the RVC device for testing, SVE, and general reference. 

At a minimum, this should implement the RVC Run, RVC Clean and RVC Operational State clusters such that all the supported commands for a typical RVC device can be tested. See the example below for a simple state machine that will be sufficient.

Simplified_rvc_app_example

### PR DESCRIPTION
**Pull request by hicklin**
This PR is the same as #28719. There was something strange going on with the CI tests so we are trying to create another PR to see if it will fix things.

Fixes https://github.com/project-chip/connectedhomeip/issues/28564.

This PR adds an example RVC app that integrates the RvcRunMode and RvcOperationalState clusters with logic as would be expected it a typical RVC implementation. Below is a visual representation of the state machine of the RVC example app.

rvc-app-state-machine

### CODE REVIEW

**Review by plauric — APPROVED** [review]

File: examples/rvc-app/linux/include/CHIPProjectAppConfig.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *          to the default configuration values for general CHIP features.
+ *
+ */
+
+#pragma once
+
+// include the CHIPProjectConfig from config/standalone
+#include <CHIPProjectConfig.h>
+
+#define CHIP_DEVICE_CONFIG_DEVICE_TYPE 116 // 0x0074 = 117 = Matter Robotic Vacuum Cleaner
```
Comment (bzbarsky-apple) [inline examples/rvc-app/linux/include/CHIPProjectAppConfig.h:33]:
```suggestion
#define CHIP_DEVICE_CONFIG_DEVICE_TYPE 116 // 0x0074 = 116 = Matter Robotic Vacuum Cleaner
```

File: examples/rvc-app/rvc-common/include/rvc-device.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#include "rvc-mode-delegates.h"
+#include "rvc-operational-state-delegate.h"
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+const EndpointId RvcRunModeEndpoint          = 1;
```
Comment (bzbarsky-apple) [inline examples/rvc-app/rvc-common/include/rvc-device.h:12]: Wouldn't it make more sense to have the endpoint be a constructor argument to the RvcDevice class?

In particular, per the device type, all these clusters should be on the _same_ endpoint, so why the multiple constants?

File: examples/rvc-app/rvc-common/include/rvc-operational-state-delegate.h [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        OperationalState::GenericOperationalState(to_underlying(OperationalState::OperationalStateEnum::kError)),
+        OperationalState::GenericOperationalState(
+            to_underlying(Clusters::RvcOperationalState::OperationalStateEnum::kSeekingCharger)),
+        OperationalState::GenericOperationalState(to_underlying(Clusters::RvcOperationalState::OperationalStateEnum::kCharging)),
+        OperationalState::GenericOperationalState(to_underlying(Clusters::RvcOperationalState::OperationalStateEnum::kDocked)),
+    };
+
+    const Clusters::OperationalState::GenericOperationalPhase mOperationalPhaseList[1] = {
+        // Phase List is null
+        OperationalState::GenericOperationalPhase(DataModel::Nullable<CharSpan>()),
```
Comment (bzbarsky-apple) [inline examples/rvc-app/rvc-common/include/rvc-operational-state-delegate.h:53]: Followup: Our representation of null phaselist is pretty broken.  I thought there was already an issue tracking this, but not finding it right now.

File: examples/rvc-app/rvc-common/include/rvc-device.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * Init all the clusters used by this device.
+     */
+    void Init();
+
+    /**
+     * Handles the RvcRunMode command requesting a mode change.
+     */
+    void HandleRvcRunChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response);
```
Comment (bzbarsky-apple) [inline examples/rvc-app/rvc-common/include/rvc-device.h:54]:
```suggestion
    void HandleRvcRunChangeToMode(uint8_t newMode, ModeBase::Commands::ChangeToModeResponse::Type & response);
```

File: examples/rvc-app/rvc-common/include/rvc-device.h [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * Handles the RvcCleanMode command requesting a mode change.
+     */
+    void HandleRvcCleanChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response);
```
Comment (bzbarsky-apple) [inline examples/rvc-app/rvc-common/include/rvc-device.h:59]:
```suggestion
    void HandleRvcCleanChangeToMode(uint8_t newMode, ModeBase::Commands::ChangeToModeResponse::Type & response);
```

File: examples/rvc-app/rvc-common/src/rvc-device.cpp [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+using namespace chip::app::Clusters;
+
+void RvcDevice::Init()
+{
+    mRunModeInstance.Init();
+    mCleanModeInstance.Init();
+    mOperationalStateInstance.Init();
+}
+
+void RvcDevice::HandleRvcRunChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
```
Comment (bzbarsky-apple) [inline examples/rvc-app/rvc-common/src/rvc-device.cpp:12]:
```suggestion
void RvcDevice::HandleRvcRunChangeToMode(uint8_t newMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
```

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/rvc-app/rvc-common/include/rvc-device.h (hunk 2 above)
Comment (hicklin) [reply examples/rvc-app/rvc-common/include/rvc-device.h]: Fixed this.

File: examples/rvc-app/rvc-common/include/rvc-device.h (hunk 4 above)
Comment (hicklin) [reply examples/rvc-app/rvc-common/include/rvc-device.h]: Renamed these.

File: examples/rvc-app/rvc-common/include/rvc-operational-state-delegate.h (hunk 3 above)
Comment (hicklin) [reply examples/rvc-app/rvc-common/include/rvc-operational-state-delegate.h:53]: [This](https://github.com/project-chip/connectedhomeip/issues/27931) is the only issue that I found that could be related to this. Is this it?

Comment (bzbarsky-apple) [reply examples/rvc-app/rvc-common/include/rvc-operational-state-delegate.h:53]: @hicklin Yes, thank you.

**Pull request merged**